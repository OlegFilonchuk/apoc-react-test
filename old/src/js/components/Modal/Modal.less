@modal-header-size: 54px;
@modal-footer-size: 60px;

@modal-margin: 10px;
@modal-tablet-margin: 100px;

@modal-body-spacing: (@modal-header-size + @modal-footer-size + (@modal-margin * 2));
@modal-body-spacing-tablet: (@modal-header-size + @modal-footer-size + (@modal-tablet-margin * 2));

// .modal-dialog class is required in order no to mix styles with the CanJS based version of Apoc library
.modal-dialog .sc-dialog {
  pointer-events: none;

  &,
  &-backdrop {
    &.sc-on {
      display: block;
    }
  }

  &-header {
    height: @modal-header-size;
  }

  &-body {
    overflow-y: auto;
    max-height: calc(100vh - @modal-body-spacing);

    @media (min-width: 768px) {
      max-height: calc(100vh - @modal-body-spacing-tablet);
    }
  }

  &-footer {
    height: @modal-footer-size;
  }

  &-backdrop {
    display: none;

    &:not(.sc-on) {
      z-index: -1;
    }
  }

  .sc-dialog-box {
    pointer-events: all;
    margin: 50px auto;
  }
}
