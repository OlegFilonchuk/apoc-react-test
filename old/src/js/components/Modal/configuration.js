/* eslint-disable import/prefer-default-export */
export const Z_INDEX = 15500;
/* eslint-enable import/prefer-default-export */
