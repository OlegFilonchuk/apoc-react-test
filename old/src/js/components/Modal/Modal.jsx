import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Z_INDEX } from './configuration';

import './Modal.less';

export default function Modal(props) {
  const dialogClass = classNames('sc-dialog', 'sc-animate', {
    'sc-on': props.isOpen
  });

  return (
    <div className="modal-dialog">
      <div className={dialogClass} style={{ zIndex: props.zIndex }}>
        <div className={`sc-dialog-box ${props.className}`}>
          <div className="sc-dialog-content">
            <div className="sc-dialog-header">
              <h3 className="sc-dialog-title">{props.header}</h3>
            </div>
            <div className="sc-dialog-body">{props.children}</div>
            <div className="sc-dialog-footer">{props.footer}</div>
          </div>
        </div>
      </div>
    </div>
  );
}

Modal.propTypes = {
  /* eslint-disable react/no-unused-prop-types */
  /**
   * Modal name for multistacking
   */
  name: PropTypes.string,
  /* eslint-enable react/no-unused-prop-types */

  /**
   * isOpen determines if modal is opened
   */
  isOpen: PropTypes.bool,

  /**
   * Header content of modal header
   */
  header: PropTypes.node,

  /**
   * Footer content of modal footer
   */
  footer: PropTypes.node,

  /**
   * ClassName size/appearance modifier class
   */
  className: PropTypes.string,

  /**
   * Children modal body content
   */
  children: PropTypes.node,

  /**
   * Modal z-index
   */
  zIndex: PropTypes.number
};

Modal.defaultProps = {
  isOpen: false,
  zIndex: Z_INDEX,
  children: '',
  className: '',
  header: '',
  footer: '',
  name: ''
};
