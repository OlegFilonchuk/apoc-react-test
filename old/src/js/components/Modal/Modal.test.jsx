import React from 'react';
import { shallow } from 'enzyme';

import Modal from './Modal';

const mountModal = (header, content, footer, isOpen, mountFn = shallow) =>
  mountFn(
    <Modal header={header} isOpen={isOpen} footer={footer}>
      {content}
    </Modal>
  );

describe('<Modal />', () => {
  it('should set closed modal', () => {
    // arrange
    const header = 'Foo title';
    const content = <p>Bar</p>;
    const footer = (
      <div>
        <button>Test</button>
      </div>
    );
    const isOpen = false;

    // act
    const component = mountModal(header, content, footer, isOpen);

    // assert
    expect(component).toMatchSnapshot();
  });

  it('should set opened modal', () => {
    // arrange
    const header = 'Foo title';
    const content = <p>Bar</p>;
    const footer = (
      <div>
        <button>Test</button>
      </div>
    );
    const isOpen = true;

    // act
    const component = mountModal(header, content, footer, isOpen);

    // assert
    expect(component).toMatchSnapshot();
  });

  it('should allow to open modal from parent component', () => {
    // arrange
    const isOpen = false;

    // act
    const component = mountModal(undefined, undefined, undefined, isOpen);

    const firstResultNodes = component.find('.sc-on');

    // assert
    expect(firstResultNodes.length).toBeFalsy();
    component.setProps({ isOpen: true });

    const secondResultNodes = component.find('.sc-on');

    expect(secondResultNodes.length).toBeTruthy();
  });
});
