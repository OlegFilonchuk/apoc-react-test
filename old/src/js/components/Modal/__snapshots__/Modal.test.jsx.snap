// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should set closed modal 1`] = `
<div
  className="modal-dialog"
>
  <div
    className="sc-dialog sc-animate"
    style={
      Object {
        "zIndex": 15500,
      }
    }
  >
    <div
      className="sc-dialog-box "
    >
      <div
        className="sc-dialog-content"
      >
        <div
          className="sc-dialog-header"
        >
          <h3
            className="sc-dialog-title"
          >
            Foo title
          </h3>
        </div>
        <div
          className="sc-dialog-body"
        >
          <p>
            Bar
          </p>
        </div>
        <div
          className="sc-dialog-footer"
        >
          <div>
            <button>
              Test
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Modal /> should set opened modal 1`] = `
<div
  className="modal-dialog"
>
  <div
    className="sc-dialog sc-animate sc-on"
    style={
      Object {
        "zIndex": 15500,
      }
    }
  >
    <div
      className="sc-dialog-box "
    >
      <div
        className="sc-dialog-content"
      >
        <div
          className="sc-dialog-header"
        >
          <h3
            className="sc-dialog-title"
          >
            Foo title
          </h3>
        </div>
        <div
          className="sc-dialog-body"
        >
          <p>
            Bar
          </p>
        </div>
        <div
          className="sc-dialog-footer"
        >
          <div>
            <button>
              Test
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
